### ISSUE TITLE
"Shuffle" in validation dataloader: is it really best practices?

### ISSUE THREAD

**Issue by jbohnslav** [PyTorchLightning/pytorch-lightning#5028]
## 🐛 Bug

In my `LightningModule`'s `val_dataloader` method, I have this dataloader: 

```python
dataloader = DataLoader(self.datasets[split], batch_size=batch_size,
                                shuffle=True, num_workers=self.hparams.compute.num_workers,
                                pin_memory=torch.cuda.is_available(), drop_last=False)
return dataloader
```

I receive this warning: 
```
.../pytorch_lightning/utilities/distributed.py:45: UserWarning: Your val_dataloader has `shuffle=True`, it is best practice to turn this off for validation and test dataloaders.
  warnings.warn(*args, **kwargs)
```
However, it's quite important for me to shuffle my validation batches. For example, I visualize the first few batches in my validation to get an idea of random model performance on my images-- without shuffling, I'd only be able to inspect the same images every epoch. 

### Expected behavior

No warning

### Additional information

This is more of a discussion than a bug report, but it didn't neatly fit into any categories. Do we really think it's important enough to warn the user when using shuffle in validation? I've tried suppressing it, but I can't figure out where exactly it's called.

**Comment by carmocca** [comment]
Wouldn't you want to always inspect the same images to properly assess the model performance?
If you keep looking at different samples each time, it is harder to know if the change in performance is caused by the model improvement or by how well the model generalizes to the samples in particular.

> I've tried suppressing it, but I can't figure out where exactly it's called.

You shouldn't need to, see this example of using `filterwarnings`

```python
import warnings

def test():
     warnings.warn("this is a test", UserWarning)

warnings.filterwarnings("ignore", category=UserWarning, message="this is a test")
test() # No warning!
```

**Comment by jbohnslav** [comment]
> Wouldn't you want to always inspect the same images to properly assess the model performance?

**Comment by bjourne** [comment]
But the issue remains. WHY does PL "strongly recommend" that you don't shuffle validation data? Afaict, the warning is completely pointless.
